### PR TITLE
Improve ranger implementation

### PR DESCRIPTION
**Pull request by ymasiuk**

### CODE REVIEW

File: app/assets/javascripts/lib/ranger_connection.js.coffee [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
+  type: "POST",
+  url: barongPath,
+  async: true,
+  xhrFields: {
+    withCredentials: true
+  },
+  success: (res) ->
+    gon.user.token = res
+    ranger.bind 'open', ->
+      console.info 'Connection to Ranger at has been established'
```
Comment (kovetskiy) [inline app/assets/javascripts/lib/ranger_connection.js.coffee]: "at has been"
It's not even a sentence :smile:

Comment (ymasiuk) [reply app/assets/javascripts/lib/ranger_connection.js.coffee]: @kovetskiy you merged this sentence in #97

File: app/assets/javascripts/lib/ranger_connection.js.coffee [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+  type: "POST",
+  url: barongPath,
+  async: true,
+  xhrFields: {
+    withCredentials: true
+  },
+  success: (res) ->
+    ranger.bind 'open', ->
+      console.info 'Connection to Ranger at has been established'
+      return if not gon.user
```
Comment (kovetskiy) [inline app/assets/javascripts/lib/ranger_connection.js.coffee]: we don't need that check anymore

File: app/assets/javascripts/lib/ranger_connection.js.coffee [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
+      console.info 'Solving authorization challenge by sending JWT token'
+      ranger.send JSON.stringify {jwt: "Bearer #{res}"}
+    ranger.bind 'error', ->
+      console.error(data)
```
Comment (kovetskiy) [inline app/assets/javascripts/lib/ranger_connection.js.coffee]: Need to add user-friendly alert

**Review by kovetskiy — CHANGES_REQUESTED** [review]
Please add alert message for the user and we merge this

File: app/assets/javascripts/lib/ranger_connection.js.coffee (hunk 2 above)
Comment (ymasiuk) [reply app/assets/javascripts/lib/ranger_connection.js.coffee]: why? if user not login we will not send call to the ranger

Comment (kovetskiy) [reply app/assets/javascripts/lib/ranger_connection.js.coffee]: Oh, I see the purpose of that check now.

**Comment by mod** [comment]
nice alert

**Pull request merged**